internal/cleanups/composite.go

```diff
@@ -12,7 +12,9 @@ type Composite struct {
 
 // Add adds a cleanup to be called.
 func (c *Composite) Add(f func(context.Context) error) {
-	c.cleanups = append(c.cleanups, f)
+	if f != nil {
```

Comment:
Wondering if it wouldn't be better to make the caller responsible for this? We might be hiding subtle bugs by just ignoring nil functions.

On the `composite.Add` we could log an error if a nil function was passed.

Reply:
> We might be hiding subtle bugs by just ignoring nil functions.

Fair 👍 

> On the composite.Add we could log an error if a nil function was passed.

I don't think this will add much value as we'd have to carrefully look at logs to spot the issue before code gets merged.